components/ListView.js

```diff
@@ -132,7 +135,18 @@ class ListView extends React.Component {
     });
 
     if (this.props.onRefresh) {
-      this.props.onRefresh();
+      const callback = this.props.onRefresh();
+      const setIdle = () => {
+        if(this._isMounted) {
```

Comment:
It because there is no way to _unsubscribe_ from promises, so `this._isMounted` check make sure that setState will not execute when component is UnMounted.